### PR TITLE
Chart labels, annotations, and priorityClass config

### PR DESCRIPTION
**Pull request by wozniakjan**
Kyma project currently uses fork of BTP Operator helm chart with few custom changes. We would like to upstream all configuration knobs and eventually deprecate the https://github.com/kyma-incubator/sap-btp-service-operator.

This is a first PR containing mainly just configurability for deployment annotations and priority class.

### CODE REVIEW

File: sapbtp-operator-charts/templates/deployment.yml [hunk 1]
```diff
@@ -20,6 +22,9 @@ spec:
         {{- $secretTls := (include (print $.Template.BasePath "/secret-tls.yml") .) -}}
         {{- $configSha := (print $configmap $secret $secretTls) | sha256sum }}
         checksum/config: {{ $configSha }}
+        {{- if .Values.manager.annotations }}
+        {{- toYaml .Values.manager.annotations | nindent 8 }}
+        {{- end }}
```
Comment (wozniakjan) [inline sapbtp-operator-charts/templates/deployment.yml:27]: Kyma project has [a couple of custom annotations](https://github.com/kyma-incubator/sap-btp-service-operator/blob/0b4b0a7508c6181f58638a6a71d6427eeb9e3435/chart/templates/deployment.yml#L25-L26)

File: sapbtp-operator-charts/templates/deployment.yml [hunk 2]
```diff
@@ -112,6 +117,9 @@ spec:
       imagePullSecrets: {{ toYaml .Values.manager.imagePullSecrets | nindent 8 }}
     {{- end }}
       terminationGracePeriodSeconds: 10
+      {{- if .Values.manager.priorityClassName }}
+      priorityClassName: {{ .Values.manager.priorityClassName }}
```
Comment (wozniakjan) [inline sapbtp-operator-charts/templates/deployment.yml:121]: kyma project also requires setting priority class for managed applications in kyma-system https://github.com/kyma-project/control-plane/pull/2035

**Comment by sTb1337** [comment]
@wozniakjan we also need priorityClassName to bet set. Can we reopen this merge or do I have to open a new one?

**Comment by wozniakjan** [comment]
@sTb1337 I would love to get this in, any chance we can get a review any time soon @kerenlahav, @pavelmaliy?

This PR has been closed for lack of activity and can not be reopened, as well as followup https://github.com/SAP/sap-btp-service-operator/pull/214.

**Comment by kerenlahav** [comment]
@wozniakjan what is the difference between this PR and [#204](https://github.com/SAP/sap-btp-service-operator/pull/214)?
are all of these labels relevant only for kyma clusters?

**Comment by wozniakjan** [comment]
hi @kerenlahav

> what is the difference between this PR and #204?

#204 and #201 are identical. New PR got open because the "PR reopen" button is disabled for me here. If you can reopen any of those that would be great, alternatively I can open another identical PR from the same branch.

> are all of these labels relevant only for kyma clusters?

I dare to say anyone using custom labels, annotations or priority class. Kyma is one user and @sTb1337 appears to be second :)

**Pull request closed without merging**